Microsoft.Toolkit.Uwp.Input.GazeInteraction/GazePointerProxy.cpp

```diff
@@ -14,8 +14,16 @@ DependencyProperty^ GazePointerProxy::GazePointerProxyProperty::get()
 GazePointerProxy::GazePointerProxy(FrameworkElement^ element)
     : _element(element)
 {
-    _loadedToken = element->Loaded += ref new RoutedEventHandler(this, &GazePointerProxy::OnPageLoaded);
-    _unloadedToken = element->Unloaded += ref new RoutedEventHandler(this, &GazePointerProxy::OnPageUnloaded);
+    auto content = Window::Current->Content;
+    auto ancestor = dynamic_cast<FrameworkElement^>(element->Parent);
+    while (ancestor != nullptr && ancestor != content)
+    {
+        ancestor = dynamic_cast<FrameworkElement^>(ancestor->Parent);
+    }
+    _isLoaded = ancestor != nullptr;
+
+    element->Loaded += ref new RoutedEventHandler(this, &GazePointerProxy::OnPageLoaded);
```

Comment:
Don't we need to still unregister these events later?

Reply:
I am checking, but I believe we do not need to unregister these events. Having Loaded="Wibble" in a abc.xaml file results in code generated in the corresponding abc.g.hpp file that neither stores the event handler reference nor unregisters the handler. I can also not see an event when I could reasonably unregister the event that does not appear too late.

Reply:
@peteams I know with C++/CX we had to unregister events otherwise they'd keep things around.  I'm not sure if that guidance has changed with C++/WinRT.  You were unregistering the events with the tokens before though, so I'm not sure if I follow why it can't still be done now?

Reply:
The previous code unregistered the events because they were expected to fire only once and C++/CX paranoia together with the expectation that they would not fire again led me to unregister them. The expectation changed when a bug was reported with cached pages. Loaded and Unloaded can legitimately fire multiple times, and this leads me to allow the registrations to persist.

In the previous version there was a definite time when they were unregistered, at the point they had been used once. In the new world where the expectation is that they are useful for multiple firings, there is no longer a definite time they can be removed.

Additionally, the XAML to C++/CX compiler demonstrates the same behavior as my code: hook and forget. Here is the code Loaded="Leakable" creates:

            ::Windows::UI::Xaml::Controls::Page^ element1 = safe_cast<::Windows::UI::Xaml::Controls::Page^>(__target);
            (safe_cast<::Windows::UI::Xaml::Controls::Page^>(element1))->Loaded += ref new ::Windows::UI::Xaml::RoutedEventHandler(this, (void (::App3::MainPage::*)
                (::Platform::Object^, ::Windows::UI::Xaml::RoutedEventArgs^))&MainPage::Leakable);

Reply:
Pete is correct here. Events need to be unregistered only if you do not want them firing anymore during the lifetime of the application. They need not be unregistered if the events need to exist through the life time of the application. 
Please approve for merge if there are no other issues.